src/System.Management.Automation/security/CatalogHelper.cs

```diff
@@ -272,51 +284,37 @@ internal static void GenerateCatalogFile(string cdfFilePath)
 
                 // navigate all the files hash entries in the .cdf file
                 IntPtr memberInfo = IntPtr.Zero;
-                try
+                IntPtr memberFile = IntPtr.Zero;
+                string fileName = string.Empty;
+                do
                 {
-                    IntPtr memberFile = IntPtr.Zero;
-                    NativeMethods.CryptCATCDFEnumMembersByCDFTagExErrorCallBack memberCallBack = new NativeMethods.CryptCATCDFEnumMembersByCDFTagExErrorCallBack(ParseErrorCallback);
-                    string fileName = string.Empty;
-                    do
-                    {
-                        memberFile = NativeMethods.CryptCATCDFEnumMembersByCDFTagEx(resultCDF, memberFile, memberCallBack, ref memberInfo, true, IntPtr.Zero);
-                        fileName = Marshal.PtrToStringUni(memberFile);
+                    memberFile = WinTrustMethods.CryptCATCDFEnumMembersByCDFTagEx(resultCDF, memberFile, ParseErrorCallback, ref memberInfo, true, IntPtr.Zero);
+                    fileName = Marshal.PtrToStringUni(memberFile);
```

Comment:
It is not critical but I wonder why we do manual marshalling if this could be in the method signature definition.

And please use named arguments for constatnts.

Reply:
This is a fairly complex method call and manual marshaling is done because you need to be able to pass in a null reference as well as a structure reference. For example memberFile will be `IntPtr.Zero` on the first call but is set as the return value and used as some sort of pointer to tell the native code where it is up to in the enumeration. We can't really have that be the structured object itself as you loose the pointer semantics known by the native code.

I've added the named arguments.